--- src/Document.C.poppler	2010-12-30 06:14:03.000000000 +0000
+++ src/Document.C	2010-12-30 06:17:17.000000000 +0000
@@ -486,6 +486,8 @@
 		#ifdef OLD_POPPLER
 			#warning Using poppler <= 0.5
 			textdump += poppler_page_get_text (page, rect);
+		#elif defined (HAVE_POPPLER_0_16_0)
+			textdump += poppler_page_get_selected_text (page, POPPLER_SELECTION_GLYPH, rect);
 		#else
 			#warning Using poppler >= 0.6
 			textdump += poppler_page_get_text (page, POPPLER_SELECTION_GLYPH, rect);
--- configure.in.poppler	2009-06-21 11:01:46.000000000 +0000
+++ configure.in	2010-12-30 06:31:13.000000000 +0000
@@ -54,6 +54,8 @@
 case $($PKG_CONFIG --modversion poppler-glib) in
 	0.5.*)
 		AC_DEFINE(OLD_POPPLER,"1","Using poppler < 0.6");;
+	0.20.*)
+		AC_DEFINE(HAVE_POPPLER_0_16_0, "1", "Define to 1 if you have Poppler version 0.16.0 or higher.");;
 esac
 
 
